### ISSUE TITLE
Many Errors after conversion to Swift3

### ISSUE THREAD

**Issue by TomMajor** [kaandedeoglu/KDCircularProgress#49]
Hello,

it is stated that master branch is now compatible with Swift 3.
I am using Xcode Version 8.0 (8A218a) and testing KDCircularProgress with a test project.
After CocoaPods pod install to version KDCircularProgress (1.5.2),
and opening the workspace file, Xcode wants to convert to Swift 3. This conversion fails with 66 errors in module KDCircularProgress.swift.

If it is compatible with Swift 3, what am I doing wrong here?
Thanks,
Tom

**Comment by jiangxiaopeng** [comment]
set(colors: .white, .cyan)  ,this error

**Comment by kaandedeoglu** [comment]
@TomMajor Can you please send me an example project with the problem please?

**Comment by TomMajor** [comment]
Hello kaandedeoglu,

thanks for the quick response.
I just create a new single view app in xcode, without any actual coding, close the project, install KDCircularProgress via CocoaPods (screenshot), open the new workspace file, then at once the Swift 3 conversion dialog pops up which leads to 66 errors, please see attached screenshot.

The thing discovered today is, if I just press save in this dialog, in the project the errors are gone somehow. I can build and run and it seems to work. The 66 errors confused me yesterday, I did not go further than, but today I did and it seems no real issue. Anyhow, this will confuse other user as well, I guess.

![bildschirmfoto 2016-10-09 um 16 02 54](https://cloud.githubusercontent.com/assets/7770659/19221181/6bce082c-8e3e-11e6-8fd8-3a885eea5a3f.png)
![bildschirmfoto 2016-10-09 um 16 04 41](https://cloud.githubusercontent.com/assets/7770659/19221182/75ba511a-8e3e-11e6-8ecb-5c37f515773f.png)

I think, in your project, the information that this is already Swift3 is missing somehow, the conversion dialog should not pop up, but this is just a guess.

Another small issue, the line
`set(colors: .white, .cyan)`
is not accepted in xcode, I have to remove 'colors:'
Same thing in the example code in view controller
`progress.set(colors: UIColor.cyan ,UIColor.white, UIColor.magenta, UIColor.white, UIColor.orange)`

Thanks,

**Comment by kaandedeoglu** [comment]
@TomMajor I've just ran the same scenario: 

- Created a single view app, I called it `KDTest`
- Went to the project directory and did `pod init`
- Added `pod KDCircularProgress` to `Podfile`
- `pod install`
- Open the workspace
- `import KDCircularProgress` in `ViewController.swift`
- Hit build

and everything worked smoothly. 

So I'm confused and interested :) It's either my Xcode or yours that's overriding some default setting value. I'd be perfect if a third party also ran the scenario and wrote their results.

**Comment by TomMajor** [comment]
Hello kaandedeoglu,

I did not even do your last two steps, just opening the workspace is enough.
So you don't get the Swift conversion dialog at all after opening the workspace?
If I look for Xcode updates, it seems my version 8.0 8A218a should be the latest one, strange.
Could try to make a little video to demonstrate in the next days.

What about the other issue, do you have it? set(colors: .white, .cyan)
It seems jiangxiaopeng has this issue too.

**Comment by JDandini** [comment]
The thing is: you should install Cocoapods 1.1.1 and re create the podfile usin 
```
$pod init 
$pico Podfile 
//add pod 'KDCircularProgress'
// add all your pods
//then ctl+x and save the changes on the Podfile 
$pod install 
``` 
This was happening to me but creating all with Cocoapods 1.1.1 and re create the podfile solve the issue

**Comment by TomMajor** [comment]
@JDandini thanks for the hint.
Updating CocoaPods to 1.1.1 (from 1.0.1) indeed solved the problem here. The conversion dialog in xcode with the errors is completely gone now. 
Reading the changelog for CocoaPods,
https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md
this could be this fix -> "Use Swift 3.0 by default during validation."

Also the other small issue I wrote about, the line 
`set(colors: .white, .cyan)`
is accepted in xcode again.

Thanks to JDandini for the hint and to kaandedeoglu for a great framework!

**Comment by kaandedeoglu** [comment]
so glad it's useful for you. and sorry i couldn't be of more help:)
